packages/icons/src/Icon.js

```diff
@@ -10,29 +10,32 @@ const Icon = ({ name, title, desc, titleId, descId, ...props }) => {
   ariaLabelledBy += desc && descId ? ` ${descId}` : ''
   ariaLabelledBy = ariaLabelledBy ? ariaLabelledBy : undefined
 
-  return (
-    Component && (
+  if (!Component) {
+    if (process.env.NODE_ENV !== 'production')
+      console.trace(`icon ${iconName} does not exist`)
+    return null
+  } else {
+    return (
       <Component
         title={title}
         desc={desc}
         titleId={titleId}
         descId={descId}
-        aria-hidden={!!ariaLabelledBy}
+        aria-hidden={!!ariaLabelledBy || true}
```

Comment:
Do you think it would make more sense to change `undefined` on line 11 to true?

Reply:
yea, this seems incorrect. `X || true` is always truthy, and `!!` already coerced `ariaLabelledBy` to bool

Reply:
@James300 / @sdalonzo,
 
Think there's a misunderstanding. At least here what I understood, which could be an overlook. Seems that the current version is correct logic to generate HTML attribute `aria-hidden="true"`, if `ariaLabelledBy` is invalid. Otherwise, `aria-hidden="desc-id-value"`.

Confirmed the logic w. @dgiraldo313 when doing the PR to ensure the a11y requirement. (ping @dgiraldo313 to review the PR, and chime in if he has any thoughts on this logic. Let me know, if I misunderstood your comments. Happy to colla. to get to the bottom of this thread.

Here's the logic from my interpretation.

1) `ariaLabelledBy` variant is determined by`titleId`, `desc` & `descId`. If any of those doesn't validate the rules, default value `undefined` is assigned to `ariaLabelledBy` variant:

https://github.com/pricelinelabs/design-system/blob/39beb570538f22489424b2f3ce355dc7c7cd8b37/packages/icons/src/Icon.js#L9-L11

2. In this update, the PR removed the default prop `aria-hidden: true`, and apply as fall-back default value `true`, if `ariaLabelledBy` from step 1 is ever invalid w. the intentional `undefined` value. This works because `!!undefined || true` => `false || true` => `true` => generate HTML attribute `aria-hidden="true"`.

    If there's a valid `ariaLabelledBy` value, the logic would be `!!'desc-id' || true` => generate HTML attribute `aria-hidden="desc-id"` for valid a11y.

```jsx
aria-hidden={!!ariaLabelledBy || true}
```

Reply:
@BeniCheni except that the `!!` already coerced the `aria-labelled-by` to a bool:

![image](https://user-images.githubusercontent.com/2132853/66757312-e86dd680-ee69-11e9-9a46-834e73af56f5.png)